client/engine/chainservice/simulated_backend_chainservice.go

```diff
@@ -25,7 +25,7 @@ type binding[T any] struct {
 	Contract *T
 }
 
-type bindings struct {
+type Bindings struct {
```

Comment:
This is a type accepted by `NewSimulatedBackendChainService` so it should probably be public? Keen for feedback here.

Reply:
There's a bit of a circle going here - the bindings are produced as output from `SetupSimulatedBackend`. Having it private means that external packages can't declare literals of this type, but it's not clear why they'd want or need to manually construct these bindings.

The member properties are all public and accessible from outside packages, although never accessed anywhere.

I don't think this change helps or hurts in any large amount.

Reply:
Current usage all looks like `SetupSimulatedBackend()` being called, and its outputs being fed immediately into `NewSimulatedBackendChainService()`. The functions might be better as one function, which returns a chainservice for all accounts as well?